ml-agents/mlagents/plugins/stats_writer.py

```diff
@@ -31,6 +31,7 @@ def get_default_stats_writers(run_options: RunOptions) -> List[StatsWriter]:
         TensorboardWriter(
             checkpoint_settings.write_path,
             clear_past_data=not checkpoint_settings.resume,
+            hidden_keys=["Is Training", "Step"],
```

Comment:
Not a fan of the magic strings here. I think these would be better as a static frozenset() in TensorBoardWriter. 
Any thoughts?

Reply:
I encouraged @mahon94 to remove it as a magic string from the TensorboardWriter. My thinking is that the writer has no responsibility for the implementation of the trainer or its stats and as such configuration/behavior should be removed from the details of said implementation. The get_default_stats_writers in stats_writer.py plugin on the otherhand is more closely related to the main entry-point/execution of the trainer. et

If it wasn't mutable, I'd say this should just be the default value of TensorboardWriter and call it a day. Transforming it from a list to a frozenset resolves the mutability issue -- even if it's still a collection of magic strings in the writer. Do you have opinions on the coupling implications?

Reply:
I don't have a strong opinion. I am okay leaving it as is.